### DISCUSSION TITLE
REST Api request sent each time a text field is filled

### DISCUSSION THREAD

**Discussion by vincedsi340** [Budibase/budibase#9542] [Help]
Hello,

May I ask you some help, I have a Form with some text fields and a data provider linked to a REST API query with some bindings.  When testing, the query is sent each time the focus is changing, when I press TAB or when I click on the next field as if it was an onchange event (but the onchange event is not set on the textfields), and I would like form data to be sent only when all fields are completed and pressing the button.
The validation button has an action parameter to send the REST API Query with form fields values.

Here is the application tree in attachment, could you help me ?

![CaptureBudibase1](https://user-images.githubusercontent.com/124553477/216896776-8e2477a1-4338-4964-8b0f-1d9f2a3bcf08.png)

Regards,

**Comment by melohagan** [comment]
Hey @vincedsi340 

If you are using form field bindings in the Data provider (such as in filters) then anytime those fields change the data provider will be refreshed.
